### PR TITLE
Allows borgs to use ranged rightclicks

### PR DESCRIPTION
**Pull request by Maia-J**
## About The Pull Request

No idea why this is all copypaste snowflake code but here we are. Borgs can now use ranged rightlicks just like anyone else.

I honestly have barely any idea as to what I am doing, does anyone have a diagram of the attack chain? But I can assure you, it does work.

## Why It's Good For The Game

Future borg parity with EVERYTHING ELSE. Will make crushers and other weapons with rightclick functionality work if one decides to add them to borgos. Even if not, it's good to keep the parity.

## Changelog
:cl:
code: Borgo rightclick ranged code
/:cl:

### CODE REVIEW

**Comment by MrMelbert** [comment]
<Details>

<Summary> Click Chain Diagram </Summary>

chain2

</DetailsL

File: code/_onclick/cyborg.dm [hunk 1]
```diff
@@ -83,6 +83,15 @@
...
+			if(LAZYACCESS(modifiers, RIGHT_CLICK))
+				var/after_attack_secondary_result = W.afterattack_secondary(A, src, FALSE, params)
+
+				if(after_attack_secondary_result == SECONDARY_ATTACK_CALL_NORMAL)
+					W.afterattack(A, src, FALSE, params)
+			else
+				W.afterattack(A, src, FALSE, params)
+
 		if(isturf(A) || isturf(A.loc))
 			W.afterattack(A, src, 0, params)
```
Comment (MrMelbert) [inline code/_onclick/cyborg.dm:96]: This line looks suspicious now

**Comment by Majkl-J** [comment]
> Click Chain Diagram
> chain2
> 
> </DetailsL

I think I just had a stroke

File: code/_onclick/cyborg.dm (hunk 1 above)
Comment (Majkl-J) [reply code/_onclick/cyborg.dm]: I updated the check but honestly it's probably not needed anymore? _I barely know what I'm doing_

File: code/_onclick/cyborg.dm [hunk 2]
```diff
@@ -79,12 +79,18 @@
...
-		if(isturf(A) || isturf(A.loc))
-			W.afterattack(A, src, 0, params)
+		else
+			if(LAZYACCESS(modifiers, RIGHT_CLICK))
+				var/after_attack_secondary_result = W.afterattack_secondary(A, src, FALSE, params)
+
+				if(after_attack_secondary_result == SECONDARY_ATTACK_CALL_NORMAL)
+					W.afterattack(A, src, FALSE, params)
+			else if(isturf(A) || isturf(A.loc))
+				W.afterattack(A, src, FALSE, params)
```
Comment (MrMelbert) [inline code/_onclick/cyborg.dm:93]:
```suggestion
		else if(isturf(A) || isturf(A.loc))
			if(LAZYACCESS(modifiers, RIGHT_CLICK))
				var/after_attack_secondary_result = W.afterattack_secondary(A, src, FALSE, params)

				if(after_attack_secondary_result == SECONDARY_ATTACK_CALL_NORMAL)
					W.afterattack(A, src, FALSE, params)
			else 
				W.afterattack(A, src, FALSE, params)
```

File: code/_onclick/cyborg.dm (hunk 1 above)
Comment (MrMelbert) [reply code/_onclick/cyborg.dm]: Well it's prevent borgs from afterattacking atoms in their contents, so who knows

**Review by Ghommie — APPROVED** [review]

**Pull request merged**